packages/theme/src/cli/services/push.ts

```diff
@@ -99,57 +104,72 @@ export interface PushFlags {
 
   /** Require theme check to pass without errors before pushing. Warnings are allowed. */
   strict?: boolean
+
+  /** The environment to push the theme to. */
+  environment?: string[]
 }
 
 /**
  * Initiates the push process based on provided flags.
  *
  * @param flags - The flags for the push operation.
  */
-export async function push(flags: PushFlags): Promise<void> {
+export async function push(flags: PushFlags, adminSession?: AdminSession, multiEnvironment?: boolean) {
+  const environment = flags.environment
+
+  // when push is used programmatically, we don't have an admin session, so need to create one
+  const session =
```

Comment:
The [push function is exported](https://github.com/Shopify/cli/blob/cbd28df0273fcbcec06277aee17325e527f41f4a/packages/theme/src/index.ts#L52) by [@shopify/cli](https://www.npmjs.com/package/@shopify/cli)